File: m4-h2/src/main/java/ru/yegorpilipenko/otus/spring/m4h2/controller/AuthorController.java
package ru.yegorpilipenko.otus.spring.m4h2.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import ru.yegorpilipenko.otus.spring.m4h2.model.Author;
import ru.yegorpilipenko.otus.spring.m4h2.service.AuthorService;

import java.util.Collection;

import static ru.yegorpilipenko.otus.spring.m4h2.controller.AuthorController.URL;

@RestController
@RequestMapping(URL)
@RequiredArgsConstructor
public class AuthorController {

    static final String URL = "/authors";

    private final AuthorService authorService;

    @GetMapping
    public Collection<Author> getAll() {
        return authorService.getAll();
    }

}

File: m1-h5/src/main/kotlin/ru/yegorpilipenko/otus/spring/m1h5/controller/AuthorController.kt
package ru.yegorpilipenko.otus.spring.m1h5.controller

import org.springframework.shell.standard.ShellComponent
import org.springframework.shell.standard.ShellMethod
import ru.yegorpilipenko.otus.spring.m1h5.model.Author
import ru.yegorpilipenko.otus.spring.m1h5.service.AuthorService

@ShellComponent
class AuthorController(val authorService: AuthorService) {

    @ShellMethod(value = "Show all authors.", group = "Author Commands")
    fun authors(): Collection<String>
            = authorService.findAll().map { author -> "${author.id}: ${author.firstName} ${author.lastName}" }

    @ShellMethod(value = "Find author by ID.", group = "Author Commands")
    fun author(id: Long): String {
        val author = authorService.findById(id)
        return if (author != null) "${author.id}: ${author.firstName} ${author.lastName}" else "Not found"
    }

    @ShellMethod(value = "Save author.", group = "Author Commands")
    fun authorSave(firstName: String, lastName: String): String {
        val author = authorService.save(Author(firstName = firstName, lastName = lastName))
        return "${author.id}: ${author.firstName} ${author.lastName}"
    }

    @ShellMethod(value = "Remove author by ID.", group = "Author Commands")
    fun authorRemove(id: Long): String = if (authorService.removeById(id)) "Removed" else "Not found"

}
